platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/ModuleConfigAdapter.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.local;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import org.eclipse.che.api.core.model.workspace.ModuleConfig;
+import org.eclipse.che.api.workspace.server.model.impl.ModuleConfigImpl;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Dmitry Shnurenko
+ */
+public class ModuleConfigAdapter implements JsonDeserializer<ModuleConfig>, JsonSerializer<ModuleConfig> {
+
+    @Override
+    public ModuleConfig deserialize(JsonElement element, Type type, JsonDeserializationContext context) throws JsonParseException {
+        return deserializeModule(element);
+    }
+
+    private ModuleConfig deserializeModule(JsonElement element) {
+        ModuleConfigImpl config = new ModuleConfigImpl();
+        JsonObject configJson = element.getAsJsonObject();
+
+        config.setName(configJson.get("name") == null ? null : configJson.get("name").getAsString());
+        config.setPath(configJson.get("path") == null ? null : configJson.get("path").getAsString());
+        config.setType(configJson.get("type") == null ? null : configJson.get("type").getAsString());
+        config.setDescription(configJson.get("description") == null ? null : configJson.get("description").getAsString());
+        config.setMixins(deserializeMixins(configJson));
+        config.setModules(deserializeModules(configJson));
+
+        return config;
+    }
+
+    private List<ModuleConfig> deserializeModules(JsonObject configJson) {
+        List<ModuleConfig> modules = new ArrayList<>();
```

Comment:
In this case it would be better to define `modules` variable after `JsonArray modulesFromJson` doing this you will be able to create modules list instance with defined size `new ArrayList<>(modulesFromJson.size);`